tests/test_more.py

```diff
@@ -2936,3 +2937,57 @@ def test_multiple(self):
         actual = list(mi.map_except(int, iterable, ValueError, TypeError))
         expected = [0, 1, 2, 4]
         self.assertEqual(actual, expected)
+
+
+class SampleTests(TestCase):
+
+    def test_length(self):
+        """Check that *k* elements are sampled."""
+        data = [1, 2, 3, 4, 5]
+        for k in [0, 3, 5, 7]:
+            sampled = mi.sample(data, k=k)
+            actual = len(sampled)
+            expected = min(k, len(data))
+            self.assertEqual(actual, expected)
+
+    def test_samling_entire_iterable(self):
+        """If k=len(iterable), the sample contains the original elements."""
+        data = ["a", 2, "a", 4, (1, 2, 3)]
+        k = len(data)
+        actual = set(mi.sample(data, k=k))
+        expected = set(data)
+        self.assertEqual(actual, expected)
+
+    def test_invariance_under_permutations_unweighted(self):
+        """The order of the data should not matter. This is a stochastic test,
+        but it will fail in less than 1 / 10_000 cases."""
+
+        # Create a data set and a reversed data set
+        data = list(range(100))
+        data_rev = list(reversed(data))
+
+        # Sample each data set 10 times
+        data_means = [mean(mi.sample(data, k=50)) for _ in range(10)]
+        data_rev_means = [mean(mi.sample(data_rev, k=50)) for _ in range(10)]
+
+        # The difference in the means should be low, i.e. little bias
+        difference_in_means = abs(mean(data_means) - mean(data_rev_means))
+        self.assertTrue(difference_in_means < 5)
```

Comment:
Did you try the "use a fixed seed" [thing](https://stackoverflow.com/a/47164461)? I think I like that better than the arbitrary threshold here.

Reply:
Addressing this in a comment below :+1: